### PR TITLE
IDE-832 Load folders from include meta data when ECL is syntax checked

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Signed-off-by: dehilsterlexis <david.dehilster@lexisnexis.com>

### CODE REVIEW

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

**Comment by dehilsterlexis** [comment]
@GordonSmith it turns out that the parsing algorithm has an expectation that all the attributes are being parsed and so the "import XXX as YYY" is not working. I need to remedy this.

**Comment by GordonSmith** [comment]
@dehilsterlexis do you need to remedy it in this PR or in the future?

**Comment by dehilsterlexis** [comment]
@GordonSmith I fixed this and pushed the solution.

**Comment by richardkchapman** [comment]
@dehilsterlexis Needs rebasing for a clean merge

**Comment by dehilsterlexis** [comment]
@richardkchapman back to you

**Comment by richardkchapman** [comment]
@GordonSmith Please review

File: comms/EclMeta.cpp [hunk 1]
```diff
@@ -111,6 +111,47 @@ void CEclMeta::Update(const std::wstring & xml)
     }
 }
 
+bool CEclMeta::LoadImports(const std::_tstring & path, const WPathVector & folders)
+{
+    bool found = false;
+
+    if (m_masterMeta.size())
+    {
```
Comment (GordonSmith) [inline comms/EclMeta.cpp]: No need for the "if" statement

File: comms/EclMeta.cpp [hunk 2]
```diff
@@ -111,6 +111,47 @@ void CEclMeta::Update(const std::wstring & xml)
...
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return found;
+}
```
Comment (GordonSmith) [inline comms/EclMeta.cpp]: This looks excessive to me (iterating through all the existing meta items) - is there not an easier way to get the "module" ID from the "path"?

File: comms/EclMeta.cpp (hunk 1 above)
Comment (dehilsterlexis) [reply comms/EclMeta.cpp]: @GordonSmith agreed

File: comms/EclMeta.cpp (hunk 2 above)
Comment (dehilsterlexis) [reply comms/EclMeta.cpp]: @GordonSmith yes, there is a better way to do it. I am working on it now.

**Comment by dehilsterlexis** [comment]
@GordonSmith for review. I redid the logic to be quicker and a bit more logical.

**Comment by dehilsterlexis** [comment]
@GordonSmith for review

File: comms/EclMeta.cpp [hunk 3]
```diff
@@ -97,6 +118,29 @@ CEclFile *CEclMeta::GetSourceFromPath(const std::wstring & path)
...
+    std::wstring dirpath = _T("");
+
+    for (WPathVector::const_iterator itr = folders.begin(); itr != folders.end(); ++itr)
+    {
+        std::_tstring pathstr = itr->first;
+        if (clib::filesystem::is_directory(pathstr)) {
+            boost::filesystem::directory_iterator end_itr;
+            for (boost::filesystem::directory_iterator itrDir(pathstr); itrDir != end_itr; ++itrDir)
+            {
+                if (clib::filesystem::is_directory(*itrDir) &&
```
Comment (GordonSmith) [inline comms/EclMeta.cpp]: Not all modules are folders...

File: comms/EclMeta.cpp [hunk 4]
```diff
@@ -97,6 +118,29 @@ CEclFile *CEclMeta::GetSourceFromPath(const std::wstring & path)
     return NULL;
 }
 
+std::_tstring CEclMeta::GetPathFromModule(const std::_tstring & module, const WPathVector & folders)
```
Comment (GordonSmith) [inline comms/EclMeta.cpp]: return boolean - and pass in target string as a ref to be filled in.

File: comms/EclMeta.cpp [hunk 5]
```diff
@@ -111,6 +155,31 @@ void CEclMeta::Update(const std::wstring & xml)
...
+        std::wstring dirpath = _T("");
+        for (EclImportMap::iterator itrImport = imports.begin(); itrImport != imports.end(); ++itrImport)
+        {
+            std::wstring module = itrImport->second.get()->GetRef();
+            if (!m_masterMeta[module])
+            {
+                std::wstring dirpath = GetPathFromModule(module, folders);
+                if (dirpath.size())
+                {
+                    m_masterMeta[module] = new CEclFolder(dirpath);
```
Comment (GordonSmith) [inline comms/EclMeta.cpp]: Not all modules are folders.

**Comment by GordonSmith** [comment]
Closing in favor of:  https://github.com/hpcc-systems/eclide/pull/459

**Pull request closed without merging**